### PR TITLE
T4967: Allow setting container hostname

### PR DESCRIPTION
**Pull request by sever-sever**
## Change Summary

Ability to set container hostname
This host name is used as /etc/hostname

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T4967

## How to test
```
set container name busybox allow-host-networks
set container name busybox host-name 'mybox'
set container name busybox image 'busybox'

```
Check hostname in container
```
vyos@r14# sudo podman exec -it busybox  cat /etc/hostname
mybox[edit]
vyos@r14# 


vyos@r14# run connect container busybox 
WARN[0000] Switching default driver from overlay2 to the equivalent overlay driver 
/ # 
/ # 
/ # hostname 
mybox
/ # 
/ # 
/ # cat /etc/hostname 
mybox/ # 

```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -113,6 +113,15 @@
               <constraintErrorMessage>Entrypoint must be ascii characters, use &amp;quot; and &amp;apos for double and single quotes respectively</constraintErrorMessage>
             </properties>
           </leafNode>
+          <leafNode name="host-name">
+            <properties>
+              <help>Container host name</help>
+              <constraint>
+                <regex>[-a-zA-Z0-9]+</regex>
+              </constraint>
+              <constraintErrorMessage>Container name must be alphanumeric and can contain hyphens</constraintErrorMessage>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: Should be `host-name` instead of `Container name`

File: interface-definitions/container.xml.in [hunk 2]
```diff
@@ -113,6 +113,15 @@
               <constraintErrorMessage>Entrypoint must be ascii characters, use &amp;quot; and &amp;apos for double and single quotes respectively</constraintErrorMessage>
             </properties>
           </leafNode>
+          <leafNode name="host-name">
+            <properties>
+              <help>Container host name</help>
+              <constraint>
+                <regex>[-a-zA-Z0-9]+</regex>
```
Comment (c-po) [inline interface-definitions/container.xml.in]: `system host-name` command uses ` <regex>[A-Za-z0-9][-.A-Za-z0-9]*[A-Za-z0-9]</regex>`

**Review by c-po — CHANGES_REQUESTED** [review]

File: interface-definitions/container.xml.in (hunk 1 above)
Comment (sever-sever) [reply interface-definitions/container.xml.in]: Fixed!

File: interface-definitions/container.xml.in (hunk 2 above)
Comment (sever-sever) [reply interface-definitions/container.xml.in]: Fixed!

**Pull request merged**